ff/handlers/nonbonded.py

```diff
@@ -184,3 +190,102 @@ def vjp_fn(*args, **kwargs):
 
         return np.array(charges, dtype=np.float64), vjp_fn
 
+class AM1CCCHandler(SerializableMixIn):
+
+    def __init__(self, smirks, params, props):
+        """
+        Parameters
+        ----------
+        smirks: list of str (P,)
+            SMIRKS patterns for each pattern
+
+        params: np.array, (P,)
+            normalized charge for each
+
+        """
+        
+        assert len(smirks) == len(params)
+
+        self.smirks = smirks
+        self.params = np.array(params, dtype=np.float64)
+        self.props = props
+
+    def parameterize(self, mol):
+        """
+        Parameters
+        ----------
+
+        mol: Chem.ROMol
+            molecule to be parameterized.
+
+        """
+        # imported here for optional dependency
+        from openeye import oechem
+        from openeye import oequacpac
+
+        mb = Chem.MolToMolBlock(mol)
+        ims = oechem.oemolistream()
+        ims.SetFormat(oechem.OEFormat_SDF)
+        ims.openstring(mb)
+
+        for buf_mol in ims.GetOEMols():
+            oemol = oechem.OEMol(buf_mol)
+
+        result = oequacpac.OEAssignCharges(oemol, oequacpac.OEAM1Charges())
+
+        if result is False:
+            raise Exception('Unable to assign charges')
+
+        am1_charges = [] 
+        for index, atom in enumerate(oemol.GetAtoms()):
+            q = atom.GetPartialCharge()*np.sqrt(constants.ONE_4PI_EPS0)
+            am1_charges.append(q)
+
+        bond_idxs = []
```

Comment:
are bonds lexicographically unique? i.e. if a bond (a_idx, b_idx) is in bond_idxs, will you ever have a situation where (b_idx, a_idx) is also in bond_idxs? should that be allowed?

Reply:
to follow up on this, can we add an assertion at the end with:

```
unique_bond_idxs = set()
for src_idx, dst_idx in bond_idxs:
     src_idx, dst_idx = sort((src_idx, dst_idx))
     unique_bond_idxs.add((src_idx, dst_idx))
assert len(unique_bond_idxs) == len(bond_idxs)
```

Reply:
disregard this - this isn't needed after discussing offline.